resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -373,20 +377,17 @@ private[spark] class Client(
       .map(CompressionUtils.createTarGzip(_))
   }
 
-  private def getDriverLauncherService(
-      k8ClientConfig: Config,
-      kubernetesMaster: String): KubernetesSparkRestApi = {
+  private def getDriverLauncherService(k8ClientConfig: Config): KubernetesSparkRestApi = {
```

Comment:
Does the library give a helpful error message though? It would be good if we could make the error reporting more precise and give back the other options (e.g. "You should specify the master in your config file, or provide it in spark.master")